src/cmap/auth/scram.ts

```diff
@@ -140,7 +143,10 @@ async function continueScramConversation(
 
   let processedPassword;
   if (cryptoMethod === 'sha256') {
-    processedPassword = 'kModuleError' in saslprep ? password : saslprep(password);
+    processedPassword =
+      'kModuleError' in deps.saslprep || typeof deps.saslprep !== 'function'
```

Comment:
interesting, now that we validate saslprep is a function  the kModuleError check is redundant b/c if kModuleError exists the type will not be function. 

We don't have to change it, esp not with the goals here, just noting